### PR TITLE
`Reaction.product` now accepts a list of `Species`

### LINKED ISSUE #702: `Reactions.product` should accept a list
It would be interesting to allow for more complex reactions such as:

A + B <-> C + D

Therefore, the Reaction class should be modified to accept a list of products.

### PR DESCRIPTION
**Pull request by RemDelaporteMathurin**
## Proposed changes

This PR fixes #702 by allowing users to give a list of `Species` for the product of a `Reaction`.

This gives us the ability to implement reactions of the type:

A + B <-> C + D

Need to merge #703 first since the tests use some of the functionality....

## Types of changes

What types of changes does your code introduce to FESTIM?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

- [ ] Black formatted
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by RemDelaporteMathurin** [comment]
@jhdark this is now ready for review

File: test/test_complex_reaction.py
Comment (jhdark) [inline test/test_complex_reaction.py:1]: Does this need its own file or can it go into the test_reaction file?

File: test/test_reaction.py
Comment (jhdark) [inline test/test_reaction.py:1]: Would it be worth adding another test where we have A -> B + C?

**Review by jhdark — APPROVED** [review]
This looks great to me, just some small suggestions but not necessary changes

File: test/test_reaction.py
Comment (RemDelaporteMathurin) [reply test/test_reaction.py:1]: This kind of reaction is currently not implemented

File: test/test_complex_reaction.py
Comment (RemDelaporteMathurin) [reply test/test_complex_reaction.py:1]: I'd rather keep it seperate for now to 1) avoid super long files 2) keep things seperated until we have a discussion on how to organise tests

**Pull request merged**